fix(#1645): add error property to date picker

bdfranck opened the pull request:
**1. What this code change do:**
This PR is a fix for https://github.com/GovAlta/ui-components/issues/1645 that adds an `error` property to the date picker.

- **2. Make sure that you've checked the boxes below before you submit MR:**

- [x] I have read and followed the [contribution guidelines](../../contributing.md)
- [x] I have run a build locally.
- [x] I have ran unit tests locally.
- [x] I have tested the functionality.

**3. Which issue and JIRA item does this PR fix (optional)**

**4. Additional notes for the reviewer**

## Reviewer Checklist

- [ ] [Coding Standards](../contribution_guidelines/coding_standards.md) followed.

When review is satisfactory reviewer will merge the changes.

bdfranck commented:
Should I write unit tests for this new state?

bdfranck commented:
I also assume that we'll want to update `ui-components-docs` to reflect this change. I can look in to that.

bdfranck commented:
Tested the web-component...
![image](https://github.com/GovAlta/ui-components/assets/1479091/13f44a65-1ac3-4e30-8fac-c26e98d56864)


Tested the react-component...
![image](https://github.com/GovAlta/ui-components/assets/1479091/97e87a66-e64f-4cb0-86a9-4f9bb32b1ad9)

bdfranck commented on libs/web-components/src/components/date-picker/DatePicker.svelte:
```diff
@@ -36,6 +38,8 @@
   let _date: Date | null;
   let _showPopover: boolean = false;
 
+  $: isError = toBoolean(error);
```
Do I need to use a [reactive declaration](https://learn.svelte.dev/tutorial/reactive-declarations) here? 🤔 I'm just passing the value to the child input.

chrisolsen replied:
No, you shouldn't have to.

bdfranck commented on libs/web-components/src/components/date-picker/date-picker.spec.ts:
```diff
@@ -27,14 +27,16 @@ it("it renders", async () => {
 
   expect(popover?.getAttribute("relative")).toBe("true");
   expect(input?.getAttribute("value")).toBe(format(value, "PPP"));
+  expect(input?.getAttribute("error")).toBe("true");
```
I updated this unit test but we may want an additional test to better check the error state.

chrisolsen replied:
Ideally, the fact that the `error` prop is correctly being passed to the popover and input should be tested.

bdfranck commented:
D'oh! I forgot to update the `tsx` spec. I've updated it.

lizhuomeng71 commented:
Issues is verified on Chrome, Firefox, and edge for react and angular, for mobile and desktop site
![Screenshot 2024-04-19 114631](https://github.com/GovAlta/ui-components/assets/7528736/2db766d0-32d3-46f7-8e37-4e537f9a85c5)
![Screenshot 2024-04-19 114627](https://github.com/GovAlta/ui-components/assets/7528736/862a7024-77fb-4a38-b77b-be9b4a022604)
![Screenshot 2024-04-19 114617](https://github.com/GovAlta/ui-components/assets/7528736/2fad7059-472c-40a6-8ff5-36b6682a8c90)

bdfranck commented:
I've amended the commit and removed the reactive `isError` declaration. It works as expected:

<img width="653" alt="image" src="https://github.com/GovAlta/ui-components/assets/1479091/2894127f-d708-4a4d-84e4-4fbf876ea771">

bdfranck commented:
I've removed the `toBoolean` import and moved the input error checking into a separate test. 

@chrisolsen How would I check that the popover has received the error prop? 🤔 I'm not seeing any error props in `popover.svelte`. I haven't written tons of unit test so I may be misunderstanding your request. 😂

chrisolsen commented:
@bdfranck disregard the popover check, that was a mistype. I think originally I thought that you were only testing the error prop being set on the goa-datepicker, but I see you are check the error prop on the goa-input component, so it looks good :+1:  Sorry for the confusion.

tzuge commented:
:tada: This PR is included in version 1.17.0-alpha.43 :tada:

The release is available on:
- [npm package (@alpha dist-tag)](https://www.npmjs.com/package/@abgov/web-components/v/1.17.0-alpha.43)
- [GitHub release](https://github.com/GovAlta/ui-components/releases/tag/web-components-v1.17.0-alpha.43)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:

tzuge commented:
:tada: This PR is included in version 4.17.0-alpha.22 :tada:

The release is available on:
- [npm package (@alpha dist-tag)](https://www.npmjs.com/package/@abgov/react-components/v/4.17.0-alpha.22)
- [GitHub release](https://github.com/GovAlta/ui-components/releases/tag/react-v4.17.0-alpha.22)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:

tzuge commented:
:tada: This PR is included in version 4.20.3 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/@abgov/react-components/v/4.20.3)
- [GitHub release](https://github.com/GovAlta/ui-components/releases/tag/react-v4.20.3)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:

tzuge commented:
:tada: This PR is included in version 1.20.4 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/@abgov/web-components/v/1.20.4)
- [GitHub release](https://github.com/GovAlta/ui-components/releases/tag/web-components-v1.20.4)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:

tzuge commented:
:tada: This PR is included in version 3.0.0-alpha.4 :tada:

The release is available on:
- [npm package (@alpha dist-tag)](https://www.npmjs.com/package/@abgov/angular-components/v/3.0.0-alpha.4)
- [GitHub release](https://github.com/GovAlta/ui-components/releases/tag/angular-v3.0.0-alpha.4)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:

tzuge commented:
:tada: This PR is included in version 3.0.3 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/@abgov/angular-components/v/3.0.3)
- [GitHub release](https://github.com/GovAlta/ui-components/releases/tag/angular-v3.0.3)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
